src/Models/AZ3166Device.ts

```diff
@@ -95,10 +98,71 @@ export class AZ3166Device extends ArduinoDeviceBase {
     return true;
   }
 
+  async upload(): Promise<boolean> {
+    try {
+      if (!this.outputPath) {
+        vscode.window.showWarningMessage(
+            'No device compile output folder found.');
+        this.channel.show();
+        this.channel.appendLine('No device compile output folder found.');
+        return false;
+      }
+
+      const binFiles = fs.readdirSync(this.outputPath).filter(
+        file => path.extname(file).endsWith(constants.binFileExt) && !path.basename(file).endsWith(constants.otaBinFileExt));
+      // const binFiles = fs.listSync(this.outputPath, ['bin']);
+      if (!binFiles || !binFiles.length) {
+        const message =
+            'No bin file found. Please run the command of Device Compile first.';
+        vscode.window.showWarningMessage(message);
+        this.channel.show();
+        this.channel.appendLine(message);
+        return false;
+      }
+
+      try {
+        const hostVolumes = 
+        await sdk.FileSystem.listVolume();
+        const AZ3166Disk = hostVolumes.find(volume => {
+          const nameProperty = Object.getOwnPropertyDescriptor(volume, 'name');

```

Comment:
We can simply use `volume => volume.name === 'AZ3166'` here

Reply:
Got.